elastic/kibana
# [Synthetics] adjust run_from.id for private locations

## Release Note

All Synthetics errors that are in-progress for private locations at the time the stack is upgrading will be resolved. A new error state will take its place.

## Summary

Resolves https://github.com/elastic/kibana/issues/156320

Adjust the value of the `run_from.id` field.

Before
Both the `run_from.id` and the `run_from.name` were the same value.

After
<img width="543" alt="Screen Shot 2023-05-01 at 7 55 24 PM" src="https://user-images.githubusercontent.com/11356435/235551964-bb9c419d-2ada-4a39-bd08-ce6f5021747b.png">

### Testing
1. Check out this PR from integrations https://github.com/elastic/integrations/pull/6047
2. In the integrations repo `cd packages/synthetics`
3. Run `elastic-package clean` then `elastic-package build` (You may need to upgrade elastic package if you're significantly behind)
4. In your `kibana.dev.yml` file add `xpack.fleet.registryUrl: https://localhost:8080`
5. Start kibana with `NODE_EXTRA_CA_CERTS=$HOME/.elastic-package/profiles/default/certs/kibana/ca-cert.pem yarn start --no-base-path`
6. Create a private location
7. Create a monitor assigned to that private location
8. Inspect the agent policy for that private location, then find the integration policy for that monitor. Ensure the `run_from.id` is the id of the location, and the `run_from.name` is the name of the location.

(If you run into problems setting up your local kibana connected to elastic-package registry, see https://github.com/elastic/security-team/blob/main/docs/cloud-security-posture-team/kibana/local-setup-using-elastic-package.mdx)
